pkg/deploy/registry/deployconfig/etcd/etcd.go

```diff
@@ -46,3 +74,140 @@ func NewStorage(s storage.Interface) *REST {
 
 	return &REST{store}
 }
+
+// ScaleREST contains the REST storage for the Scale subresource of DeploymentConfigs.
+type ScaleREST struct {
+	registry     *deployconfig.Registry
+	rcNamespacer kclient.ReplicationControllersNamespacer
+}
+
+// ScaleREST implements Patcher
+var _ = rest.Patcher(&ScaleREST{})
+
+// New creates a new Scale object
+func (r *ScaleREST) New() runtime.Object {
+	return &extensions.Scale{}
+}
+
+func (r *ScaleREST) Get(ctx kapi.Context, name string) (runtime.Object, error) {
+	deployment, err := (*r.registry).GetDeploymentConfig(ctx, name)
+	if err != nil {
+		return nil, errors.NewNotFound("scale", name)
+	}
+
+	// TODO(directxman12): this is going to be a bit out of sync, since we are calculating it
+	// here and not as part of the deploymentconfig loop -- is there a better way of doing it?
+	totalReplicas, err := r.replicasForDeploymentConfig(deployment.Namespace, deployment.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	var specReplicas int
+
+	// current replicas reflects either the scale of the current deployment,
+	// or the scale of the RC template if no current deployment exists
+	controller, err := r.rcNamespacer.ReplicationControllers(deployment.Namespace).Get(util.LatestDeploymentNameForConfig(deployment))
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			specReplicas = deployment.Template.ControllerTemplate.Replicas
+		} else {
+			return nil, err
+		}
+	} else {
+		specReplicas = controller.Spec.Replicas
+	}
+
+	return &extensions.Scale{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:              name,
+			Namespace:         deployment.Namespace,
+			CreationTimestamp: deployment.CreationTimestamp,
+		},
+		Spec: extensions.ScaleSpec{
+			Replicas: specReplicas,
+		},
+		Status: extensions.ScaleStatus{
+			Replicas: totalReplicas,
+			Selector: deployment.Template.ControllerTemplate.Selector,
+		},
+	}, nil
+}
+
+func (r *ScaleREST) Update(ctx kapi.Context, obj runtime.Object) (runtime.Object, bool, error) {
+	if obj == nil {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("nil update passed to Scale"))
+	}
+	scale, ok := obj.(*extensions.Scale)
+	if !ok {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("wrong object passed to Scale update: %v", obj))
+	}
+	deployment, err := (*r.registry).GetDeploymentConfig(ctx, scale.Name)
+	if err != nil {
+		return nil, false, errors.NewNotFound("scale", scale.Name)
+	}
+
+	useDC := false
+
+	// this tries to update the current deployment RC first, since updating
+	// Replicas on a DeploymentConfig doesn't do anything after the first deployment
+	// has been made
+	controller, err := r.rcNamespacer.ReplicationControllers(deployment.Namespace).Get(util.LatestDeploymentNameForConfig(deployment))
+	if err != nil {
+		if kerrors.IsNotFound(err) {
```

Comment:
Please try to avoid nesting and `else` clauses. This can be boiled down to:
```go
if err != nil {
    if !kerrors.IsNotFound(err) {
        return nil, false, err
    }
    useDC = true
}
```